readthedocsredirect.py

```diff
@@ -19,41 +19,118 @@
 
-# Setup a redirect
-response = requests.post(
-    f"https://readthedocs.org/api/v3/projects/{args.project}/redirects/",
-    json=dict(from_url="/en/latest/$rest", to_url=github, type="exact"),
-    headers=HEADERS,
-)
-assert response.ok, response.json()
+        self.project = project
+        self.github_url = f"https://{org}.github.io/{repo}/master/"
+        self.project_url = f"https://readthedocs.org/api/v3/projects/{project}"
+        self.read_the_docs_url = f"https://{project}.readthedocs.io"
+
+    def link_project_to_github(self):
+        response = requests.get(
+            self.project_url, headers=self.HEADERS
+        )
+        assert response.ok, response.json()
+
+        data = response.json()
+        data.update(
+            {
+            "default_branch": self.project,
+            "repository": {"url":  "https://github.com/dls-controls/readthedocsredirect"},
+            "homepage": self.github_url,
+            }
+        )
+        response = requests.patch(
+            self.project_url,
+            json=data,
+            headers=self.HEADERS,
+        )
+        assert response.ok, response.json()
+
+    def check_redirects(self) -> Set[str]:
+        """
+        Checks for redirects already set up through this script.
+        Errors if redirects are present which aren't expected
+        """
+
+        response = requests.get(
+            f"{self.project_url}/redirects/",
+            headers=self.HEADERS
+        )
+        assert response.ok, response.json()
+
+        already_redirected = set()
+
+        for redirect in response.json()["results"]:
+            if redirect["to_url"] != self.github_url:
+                raise RuntimeError(
+                    "Project contains an unknown redirect "
+                    f"{redirect['from_url']} to {redirect['to_url']}."
+                    "Delete it before proceeding."
+                )
+            already_redirected.add(redirect["from_url"].split("/")[2])
+
+        return already_redirected
+
+    def get_active_versions(self) -> Set[str]:
+        # Set a redirect on all active versions
+        response = requests.get(
+            f"{self.project_url}/versions", headers=self.HEADERS
+        )
+        assert response.ok, response.json()
+        versions = response.json()["results"]
+
+        return set(version["verbose_name"] for version in versions if version["active"])
+
+    def set_up_redirects_on_versions(self, versions: Set[str]):
+        for version_name in versions:
+            # Setup a redirect for 
+            response = requests.post(
+                f"{self.project_url}/redirects/",
+                json=dict(from_url=f"/en/{version_name}/$rest", to_url=self.github, type="exact"),
+                headers=self,
+            )
+            assert response.ok, response.json()
+
+    def redirect(self):
+        print("linking project to https://github.com/dls-controls/readthedocsredirect")
```

Comment:
This log message is a bit hard to understand for a user. It's probably unnecessary for a user of this script to know what it's doing under the covers. So perhaps replace this with a more generic "Beginning reconfiguration of readthedocs project {project}" or somesuch